src/main/java/bio/terra/cli/service/WorkspaceManagerServiceAws.java

```diff
@@ -50,6 +50,56 @@ public static WorkspaceManagerServiceAws fromContext() {
         Context.requireUser().getTerraToken(), Context.getServer());
   }
 
+  private void deleteControlledAwsResource(UUID workspaceId, UUID resourceId) {
+    String asyncJobId = UUID.randomUUID().toString();
+    DeleteControlledAwsResourceRequestBody deleteRequest =
+        new DeleteControlledAwsResourceRequestBody().jobControl(new JobControl().id(asyncJobId));
+
+    handleClientExceptions(
+        () -> {
+          ControlledAwsResourceApi controlledAwsResourceApi =
+              new ControlledAwsResourceApi(apiClient);
+          // make the initial delete request
+          HttpUtils.callWithRetries(
+              () ->
+                  controlledAwsResourceApi.deleteAwsS3StorageFolder(
```

Comment:
Should this be checking resource type before calling this? This common method will only work with s3 storage folders as is right? Or are we adding cases for other resources as we add them?

Reply:
It will be used with sagemaker notebooks as well part of https://github.com/DataBiosphere/terra-cli/pull/474

Reply:
Reverted this change. 
initial plan was to use a switch case and compare the resource type (similar to implementation in WSM). 
removed it since it added additional look ups and provided minimal benefit